RoutesApp_ios/modules/admin/lineRoute/controllers/LineRouteEditModeViewController.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  LineRouteEditModeViewController.swift
+//  RoutesApp_ios
+//
+//  Created by Alvaro Choque on 31/10/22.
+//
+
+import UIKit
+import FirebaseFirestore
+import SVProgressHUD
+
+class LineRouteEditModeViewController: UIViewController {
+
+    @IBOutlet weak var nameTextField: UITextField!
+    @IBOutlet weak var avgVelocityTextField: UITextField!
+    @IBOutlet weak var colorPreviewView: UIView!
+    @IBOutlet weak var finishButton: UIButton!
+
+    let line: Lines
+    var currLineRoute: LineRouteInfo?
+    let viewmodel = LineRouteEditModeViewModel()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupViews()
+        initViewModel()
+    }
+
+    init(line: Lines, targetLineRoute: LineRouteInfo? = nil) {
+        self.line = line
+        self.currLineRoute = targetLineRoute
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError()
+    }
+
+    @IBAction func chooseColor(_ sender: Any) {
+        let picker = UIColorPickerViewController()
+        picker.delegate = self
+        self.present(picker, animated: true, completion: nil)
+    }
+
+    @IBAction func finish(_ sender: Any) {
+        guard let name = nameTextField.text, let avgVel = avgVelocityTextField.text else {return}
+        SVProgressHUD.show()
+        if let targetLineRoute = currLineRoute {
+            viewmodel.editLineRoute(targetLineRoute: targetLineRoute, newName: name,
+                                    newAvgVel: avgVel,
+                                    newColor: colorPreviewView.backgroundColor?.toHexString() ?? ConstantVariables.primaryColorHexValue,
+                                    newRoutePoitns: targetLineRoute.routePoints,
+                                    newStops: targetLineRoute.stops,
+                                    newStart: targetLineRoute.start, newEnd: targetLineRoute.end)
+        } else {
+            guard let dummyLat = viewmodel.cityCoords?.latitude,
```

Comment:
if this dummy information is here to be updated in the future, add a TODO: comment on top, if not, change the name as dummy sounds like test information.